test/widgets/reset_password_page_test.dart

```diff
@@ -0,0 +1,192 @@
+      of: find.byKey(const Key('resetPasswordPasswordField')),
+      matching: find.byType(TextFormField),
+    );
+    final Finder passwordConfirmFieldFinder = find.descendant(
+      of: find.byKey(const Key('resetPasswordPasswordConfirmField')),
+      matching: find.byType(TextFormField),
+    );
+    final Finder resetPasswordButtonFinder = find.byType(LoadingButton);
+
+    testWidgets('Shows the password form', (WidgetTester tester) async {
+      await pumpMaterial(tester, ResetPasswordPage(onPasswordReset: onPasswordReset));
+
+      final Finder resetPasswordFormFinder = find.byType(ResetPasswordForm);
+      final ResetPasswordFormState formState = tester.state(resetPasswordFormFinder);
+
+      expect(formState.buttonState, ButtonState.idle);
+
+      expect(passwordFieldFinder, findsOneWidget);
+      expect(passwordConfirmFieldFinder, findsOneWidget);
+    });
+
+    testWidgets('Validates the password', (WidgetTester tester) async {
+      await pumpMaterial(tester, ResetPasswordPage(onPasswordReset: onPasswordReset));
+
+      final ResetPasswordFormState formState = tester.state(find.byType(ResetPasswordForm));
+
+      final FormFieldState passwordField = tester.state(passwordFieldFinder);
+
+      // Not validated yet, so no error
+      expect(passwordField.hasError, isFalse);
+
+      formState.formKey.currentState!.validate();
+      await tester.pumpAndSettle();
+      // Empty, so error
+      expect(passwordField.hasError, isTrue);
+
+      await tester.enterText(passwordFieldFinder, 'abcdefg');
+      formState.formKey.currentState!.validate();
+      await tester.pumpAndSettle();
+      // Too short, so error
+      expect(passwordField.hasError, isTrue);
+
+      await tester.enterText(passwordFieldFinder, 'abcdefgh');
+      formState.formKey.currentState!.validate();
+      await tester.pumpAndSettle();
+      // No number, so error
+      expect(passwordField.hasError, isTrue);
+
+      await tester.enterText(passwordFieldFinder, 'abcdefg0');
+      formState.formKey.currentState!.validate();
+      await tester.pumpAndSettle();
+      // No capital letters, so error
+      expect(passwordField.hasError, isTrue);
+
+      await tester.enterText(passwordFieldFinder, 'ABCDEFG0');
+      formState.formKey.currentState!.validate();
+      await tester.pumpAndSettle();
+      // No lower letters, so error
+      expect(passwordField.hasError, isTrue);
+
+      await tester.enterText(passwordFieldFinder, 'abcdEFG0');
+      formState.formKey.currentState!.validate();
+      await tester.pumpAndSettle();
+      // No special, so error
+      expect(passwordField.hasError, isTrue);
+
+      await tester.enterText(passwordFieldFinder, password);
+      formState.formKey.currentState!.validate();
+      await tester.pumpAndSettle();
+      expect(passwordField.hasError, isFalse);
+    });
+
+    testWidgets('Validates the password confirmation', (WidgetTester tester) async {
```

Comment:
again can you refactor this

Reply:
No here I really don't know how to refactor the tests. It's not just one widget, it's the combination of multiple widgets.

The only way to do this would be to move "PasswordField + PasswordConfirmationFIeld" into a new widget, that could then be tested. But as long as the original code is duplicated, the test needs to be duplicated as well.